src/controller.cpp

```diff
@@ -344,7 +344,12 @@ qint64 Controller::connectionTimestamp() const {
     case Controller::State::StateSilentSwitching:
       [[fallthrough]];
     case Controller::State::StateSwitching:
-      return m_connectedTimeInUTC.toMSecsSinceEpoch();
+      if (m_connectedTimeInUTC.isValid()) {
+        return m_connectedTimeInUTC.toMSecsSinceEpoch();
+      }
+      const QDateTime& initialTimeStampForExtension =
```

Comment:
Could we use `clearConnectedTime` or `resetConnectedTime`? Or even just set `m_connectedTimeInUTC` here if it's invalid?

Reply:
Possibly but I'm hesitant to do anything to modify the client timer given this value is only for passing to the extension.

Reply:
I really think we should at least set `m_connecteedTimeInUTC` here - if the timer doesn't currently exist (but should), I don't see a harm in starting the timer. You understand this part of the codebase better than I do though, and so feel free to merge as-is if this seems potentially harmful. I trust you.